netaddr.go

```diff
@@ -781,28 +780,20 @@ func FromStdAddr(stdIP net.IP, port int, zone string) (_ IPPort, ok bool) {
 	return IPPort{IP: ip, Port: uint16(port)}, true
 }
 
-var udpAddrPool = &sync.Pool{
-	New: func() interface{} { return new(net.UDPAddr) },
-}
-
 // UDPAddr returns a standard library net.UDPAddr from p.
 // The returned value is always non-nil. If p.IP is the zero
 // value, then UDPAddr.IP is nil.
 //
-// UDPAddr necessarily does two allocations. If you call PutUDPAddr
-// after you're done with it, though, then subsequent UDPAddr calls
-// can reuse the memory.
-func (p IPPort) UDPAddr() *net.UDPAddr {
-	ua := udpAddrPool.Get().(*net.UDPAddr)
-	ua.Port = int(p.Port)
-	ua.IP, ua.Zone = p.IP.ipZone(ua.IP)
-	return ua
-}
-
-// PutUDPAddr adds ua to an internal pool for later reuse by IPPort.UDPAddr.
-// Use of PutUDPAddr is optional; improper use can cause mysterious errors.
-// You must only call PutUDPAddr if there are no remaining references to ua.
-func PutUDPAddr(ua *net.UDPAddr) { udpAddrPool.Put(ua) }
+// UDPAddr necessarily does two allocations. If you have an existing
+// UDPAddr already allocated, see UDPAddrReuse.
```

Comment:
there is no UDPAddrReuse.

Reply:
Thanks, GitHub. I left that comment on the second commit where it didn't show up, but in the unified view my comment looks stupid.

Reply:
Sorry @bradfitz, I made this PR just gecause @josharian offered to help figure out how to get UDPAddr(reuse *net.UDPAddr) below the inlining threshold, and I was curious to see the technique at work. But if you prefer separate methods, consider this PR aborted and closed.